src/dotnet/APIView/APIViewIntegrationTests/CosmosReviewRepositoryTests.cs

```diff
@@ -0,0 +1,222 @@
+        {
+            await CreateTestDataIfNotExistAsync();
+            var result = await _cosmosReviewRepository.GetReviewsAsync(null, null, boolVal, null, false, 0, 50, "LastUpdated");
+            Assert.All(result.Reviews, item => Assert.Equal(item.IsClosed, boolVal));
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task GetReviewsOverload3_WithOnlyIsApproved_Test(bool boolVal)
+        {
+            await CreateTestDataIfNotExistAsync();
+            var result = await _cosmosReviewRepository.GetReviewsAsync(null, null, null, null, boolVal, 0, 50, "LastUpdated");
+            Assert.All(result.Reviews, item => Assert.Equal(item.IsApproved, boolVal));
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task GetReviewsOverload3_WithSearchIsClosedAndIsApproved_Test(bool boolVal)
+        {
+            await CreateTestDataIfNotExistAsync();
+            List<string> searchQueries = new List<string> {
+                "Dummy"
+            };
+            var result = await _cosmosReviewRepository.GetReviewsAsync(searchQueries, null, boolVal, null, boolVal, 0, 50, "LastUpdated");
+            Assert.Equal(16, result.Reviews.Count());
+            Assert.Equal(16, result.TotalCount);
+            Assert.All(result.Reviews, item => Assert.Equal(item.IsClosed, boolVal));
+            Assert.All(result.Reviews, item => Assert.Equal(item.IsApproved, boolVal));
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task GetReviewsOverload3_WithSearchLanguageAndFilterType_Test(bool boolVal)
+        {
+            await CreateTestDataIfNotExistAsync();
+            List<string> searchQueries = new List<string> {
+                "Dummy"
+            };
+            List<string> language = new List<string> {
+                "DummyLanguageOne"
+            };
+            List<int> filterTypes = new List<int> {
+                1
+            };
+            var result = await _cosmosReviewRepository.GetReviewsAsync(searchQueries, language, boolVal, filterTypes, boolVal, 0, 50, "LastUpdated");
+            Assert.Equal(2, result.Reviews.Count());
+            Assert.All(result.Reviews, item => Assert.Equal(ReviewType.Automatic, item.FilterType));
+        }
+
+        private async Task CreateTestDataIfNotExistAsync()
+        {
+            var dataBaseResponse = await _cosmosClient.CreateDatabaseIfNotExistsAsync("APIView");
+            var containerResponse = await dataBaseResponse.Database.CreateContainerIfNotExistsAsync("Reviews", "/id");
+
+            if (containerResponse.StatusCode.Equals(HttpStatusCode.Created))
+            {
+                foreach (int value in Enumerable.Range(1, 2))
+                {
+                    string[] languages = new String[] { "DummyLanguageOne", "DummyLanguageTwo" };
```

Comment:
Across the PR: If `Dummy` is just a placeholder word, please use an inclusive word. i.e. `test`, `placeholder`, `etc`.
More info: https://itconnect.uw.edu/work/inclusive-language-guide/#list